drivers/hmis/app/models/hmis/search/assessment_search.rb

```diff
@@ -0,0 +1,31 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+class Hmis::Search::AssessmentSearch
+  attr_accessor :input
+  def initialize(input)
+    self.input = input
+  end
+
+  def results(scope)
+    scope.
+      yield_self(&method(:with_roles))
+  end
+
+  protected
+
+  def with_roles(scope)
+    with_filter(scope, input, :roles) { scope.with_role(input.roles) }
+  end
+
+  private
+
+  def with_filter(scope, input, filter)
+    return scope unless input.respond_to?(filter) && input.send(filter)&.present?
+
+    yield
+  end
```

Comment:
Can we put this in a BaseFilter class? It seems like this would be shared across any kind of filter, if they are all set up this way.